attribute-masking-service/src/main/java/uk/gov/gchq/palisade/service/attributemask/service/KafkaHealthIndicator.java

```diff
@@ -80,23 +79,27 @@ public Health health() {
     private boolean performCheck() {
         try {
             Map<String, ConsumerGroupDescription> groupDescriptionMap = this.adminClient.describeConsumerGroups(Collections.singletonList(this.groupId))
-                            .all()
-                            .get(1, TimeUnit.SECONDS);
+                    .all()
+                    .get(1, TimeUnit.SECONDS);
 
             ConsumerGroupDescription consumerGroupDescription = groupDescriptionMap.get(this.groupId);
 
             LOGGER.debug("Kafka consumer group ({}) state: {}", groupId, consumerGroupDescription.state());
 
+
             if (consumerGroupDescription.state() == ConsumerGroupState.STABLE) {
-                return consumerGroupDescription.members().stream()
+                boolean assignedGroupPartition = consumerGroupDescription.members().stream()
                         .noneMatch(member -> (member.assignment() == null || member.assignment().topicPartitions().isEmpty()));
+                if (!assignedGroupPartition) {
+                    LOGGER.error("Failed to find kafka topic-partition assignments");
+                }
+                return assignedGroupPartition;
             }
         } catch (InterruptedException | ExecutionException | TimeoutException e) {
```

Comment:
The `InterruptedException` should result in a `Thread.currentThread.interrupt()`, but not an `ExecutionException` or `TimeoutException`

Reply:
updated